1 коммит - юнит тесты

ebatuev opened the pull request:
Создал отдельный репозиторий для юнит-тестов.
Создал Maven-проект.
Подключил Jacoco, Mockito, JUnit 4.

Проект:
1. Покрыл тестами классы Bun, Burger, Ingredient, IngredientType. 
2. Написал параметризацию в классе BunParameterTest
3. Сделан отчёт с помощью Jacoco.

Commit:
1 коммит - юнит тесты

nikolay-977 commented on src/test/java/BunParameterTest.java:
```diff
@@ -0,0 +1,34 @@
+    @Test
+    public void getPriceReturnThisPrice() {
+        Bun bun = new Bun("TestBun", price);
+        Assert.assertEquals(bun.price, bun.getPrice(), 0);
```
Необходимо исправить: в качестве ожидаемого результата следует использовать параметр price (поле класса BunParameterTest), а не поле класса Bun.

ebatuev replied:
Исправлен параметр price (поле класса BunParameterTest)

ebatuev replied:
Исправлено
+ добавлено поле тестового класса name

nikolay-977 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,18 @@
+
+    @Test
+    public void getNameReturnThisName() { // получить имя. Вернуть это имя
+        Assert.assertEquals(bun.name, bun.getName());
```
Необходимо исправить: в качестве ожидаемого результата следует использовать параметр класса BunTest, а не поле класса Bun. Либо создать локальную переменную для ожидаемого результата. Необходимо исправить во всех тестах.

ebatuev replied:
Исправлено
Создал локальные переменные для ожидаемого результата

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,64 @@
+
+    @Test
+    public void getReceipt() { // получить рецепт
+        burger.addIngredient(new Ingredient(IngredientType.SAUCE, "sour cream", 200));
```
Необходимо исправить: Тесты для класса Burger должны быть изолированы от классов Bun и Ingredient. При написании тестов для класса Burger во всех кейсах используй моки (тема: Юнит-тесты > методы mockito) и стабы (тема: Юнит-тесты > что такое стаб) классов Bun и Ingredient, а не реальные объекты.

ebatuev replied:
Исправлено

nikolay-977 commented on src/test/java/IngredientTest.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Test
+    public void getPriceReturnActualPrice() { // получить цену. Вернуть актуальную цену
+        Assert.assertEquals(10, ingredient.getPrice(), 0);
```
Необходимо исправить: не следует использовать числа, значение которых не является очевидным (антипатерн магические числа). Лучше сохранить в переменную с наименованием, отражающим суть содержимого.

ebatuev replied:
Исправлено
Создал локальные переменные

nikolay-977 commented on src/test/java/IngredientTypeTest.java:
```diff
@@ -0,0 +1,10 @@
+public class IngredientTypeTest {
+    @Test
+    public void ingredientTypeHasTwoElements() { // тип ингредиента состоит из двух элементов
+        Assert.assertEquals(2, IngredientType.values().length);
```
Также можно использовать метод IngredientType.valueOf("SAUCE").

ebatuev replied:
Добавил в тест методы для проверки
IngredientType.valueOf("SAUCE");
IngredientType.valueOf("FILLING");

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и **как запускать** (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

Commit:
2 коммит -
1) Исправлен класс BunParameterTest. Добавлено поле тестового класса name
2) Исправлен класс BurgerTest
3) Исправлен класс BunTest. Создал локальные переменные
4) Исправлен класс IngredientTest. Создал локальные переменные
5) Исправлен класс IngredientTypeTest
Добавил в тест методы для проверки
IngredientType.valueOf("SAUCE");
IngredientType.valueOf("FILLING");
